### PR TITLE
[Fix] Square root

### PR DESCRIPTION
**Pull request by d0cd**
This PR fixes a non-determinism issue in the square root circuit by ensuring that that result is less than (p - 1)/2, where p is the field modulus.

Note: this PR affects the `elligator2::encode` circuit. If determinism is not needed, we should consider adding a `square_root_unchecked` circuit.

This PR depends on #2072

### CODE REVIEW

**Review by acoglio — APPROVED** [review]

**Review by bendyarm — APPROVED** [review]
Looks good!

File: synthesizer/process/src/tests/test_credits.rs [hunk 1]
```diff
@@ -1536,9 +1536,9 @@ mod sanity_checks {
         // Compute the assignment.
         let assignment = get_assignment::<_, CurrentAleo>(stack, &private_key, function_name, &[r0, r1, r2], rng);
         assert_eq!(15, assignment.num_public());
-        assert_eq!(54688, assignment.num_private());
-        assert_eq!(54747, assignment.num_constraints());
-        assert_eq!((88514, 130701, 83646), assignment.num_nonzeros());
+        assert_eq!(55698, assignment.num_private());
+        assert_eq!(55761, assignment.num_constraints());
+        assert_eq!((90368, 132045, 84980), assignment.num_nonzeros());
```
Comment (Pratyush) [inline synthesizer/process/src/tests/test_credits.rs]: This pushes us to the next power of two here (2^18, as opposed to 2^17)

Comment (howardwu) [reply synthesizer/process/src/tests/test_credits.rs]: Great catch, I didn't notice that, thank you!

Comment (howardwu) [reply synthesizer/process/src/tests/test_credits.rs]: I've addressed the imbalance with this PR:
- #2087

**Pull request merged**